targets/TARGET_Cypress/TARGET_PSOC6/cy_sleep_api.c

```diff
@@ -41,9 +46,14 @@ void hal_deepsleep(void)
     cy_us_ticker_stop();
     cyhal_syspm_deepsleep();
     cy_us_ticker_start();
-#else
+#else // DEVICE_LPTICKER
+    cyhal_syspm_sleep();
```

Comment:
This seems like it is adding a lot of complexity to the hal_deepsleep implementation. Wouldn't it be simpler to keep the call to sleep_manager_lock_deep_sleep and let the sleep manager take care of figuring out whether to call hal_sleep or hal_deepsleep?

Reply:
`sleep_manager_lock_deep_sleep()` as used before is fine in my opinion. I changed `hal_deepsleep()` for consistency with the macro-checks in `hal_sleep()`, but if it's overly complex I can revert this bit.